lib/src/core/number_to_words/number_to_words.dart

```diff
@@ -1,78 +1,82 @@
 import 'package:persian_tools/persian_tools.dart';
 import 'package:persian_tools/src/constants/number_to_words/constants.dart';
-import 'package:persian_tools/src/internal_methods.dart';
 
-/// Gets an [int] with 3 or less digits as input and converts it to persian
-String _numToWord(int number) {
-  var unit = 100;
-  var result = '';
-
-  while (unit > 0) {
-    if ((number / unit).floor() * unit != 0) {
-      if (numberToWord.containsKey(number)) {
-        result += numberToWord[number]!;
-        break;
-      } else {
-        result += '${numberToWord[(number / unit).floor() * unit]} و ';
-        number %= unit;
-      }
-    }
-    unit = (unit / 10).floor();
-  }
-  return result;
+/// Gets an [int] as input and checks if it's a negative number
+bool _isNegative(int number) {
+  return number < 0;
 }
 
-/// Checks current [input] for negative [value] and deploying converting process
-String _convert(int number, bool ordinal) {
-  var result = <String>[];
-
-  final isNegative = number < 0;
-  number = isNegative ? number * -1 : number;
-
-  while (number > 0) {
-    result.add(_numToWord(number % base));
-    number = (number / base).floor();
-  }
-  if (result.length > 4) return '';
-
-  for (var i = 0; i < result.length; i++) {
-    if (result[i] != '' && i != 0) result[i] += ' ${scale[i]} و ';
-  }
-  result = result.reversed.toList();
-
-  var words = result.join('');
+/// Returns error message for [input] with more than 16 digits
+String _numberIsNotValidError() {
+  return errorMessage;
+}
 
-  if (words.endsWith(endsWithAnd)) {
-    words = words.substring(0, (words.length - 3));
-  }
+/// Gets an [int] as input and if exists, returns persian word
+String _getWord(int number) {
+  return numberToWord[number] ?? '';
+}
 
-  words = trim(isNegative ? 'منفی $words' : words);
+/// Gets an [String] input and add negative persian prefix to it
+String _addNegativeSuffix(String number) {
+  return 'منفی $number';
+}
 
-  if (ordinal) words = addOrdinalSuffix(words);
+/// Gets an [int] as input and returns [unitName] for numbers over 1000
+String? _getUnitName(int numberOfZeros) {
+  return numberOfZeros == 0
+      ? ''
+      : numberToWord[int.parse('1${"0" * numberOfZeros}')];
+}
 
-  return words;
+/// Gets an [int] with 3 or less digits as input and converts it to persian
+String? _transformToWord(int number) {
+  if (number == 0) return '';
+  if (number <= 9) {
+    return _getWord(number);
+  } else if (number >= 11 && number <= 19) return _getWord(number);
```

Comment:
I think we can make these three lines more efficient and readable.  Something like:

``` dart
if ( number <=19 && number!=10 ) return _getWord(number);
```

Reply:
Also, you can write comment to make it more understandable.